### ISSUE TITLE
[Regression] Element actionability fails because it is scrolled behind another element.

### ISSUE THREAD

**Issue by todd-m-kemp** [cypress-io/cypress#15618]
### Current behavior

After upgrading to Cypress 6.8.0, we are experiencing test failures as a result of elements not being able to be clicked due to being covered by another element. It seems likely that this is a regression related to changes made for #3200 .

### Desired behavior

Tests pass exactly as they did in our prior Cypress version, 6.6.0.

### Test code to reproduce

```
context('Cypress Issue', function () {

  beforeEach(function () {

    // These are not real credentials.
    // Please sign up (for free) at www.vidyard.com, verify your email address, then update these accordingly.

    const email = 'fake@email.com'
    const password = 'yourPassword'

    cy.visit('https://secure.vidyard.com/user/sign_in')
    cy.get('#username').type(email)
    cy.get('#password').type(password)
    cy.get('#sign-in').click()
    cy.wait(5000) // Just to make sure the page is fully loaded.
  })

  it('Click Issue caused by Visibility', function () {
    cy.get('[data-testid=folder-navigation]')
      .click();
  });
});
```

Fails in Cypress version 6.8.0.

![image](https://user-images.githubusercontent.com/62152998/112029160-5282b480-8b0f-11eb-930c-73c239d00eb4.png)

Passes in Cypress version 6.6.0.

![image](https://user-images.githubusercontent.com/62152998/112029419-9c6b9a80-8b0f-11eb-98e8-2686b52963b6.png)

### Additional Notes

It looks like adding the workaround proposed to avoid issue #3200 actually works around this problem too, but (at least in the case of my application) this is a worse experience because we have to use the workaround everywhere and not just in certain places as had previously been the case. 😞 

```
    cy.document().then((document) => {
      const node = document.createElement('style');
      node.innerHTML = 'html { scroll-behavior: auto !important; }';
      document.body.appendChild(node);
    });
```

### Versions

Cypress version 6.8.0 on macOS in Chrome 89 and Firefox 86.

**Comment by jennifer-shehane** [comment]
I'm not seeing this behavior in my tests. Do you have a specific viewport size I'm missing in the repro?

You can change the `scrollBehavior` option of the `cy.click()` - that is helpful is the el is ever being scrolled in a way that's not working. https://docs.cypress.io/api/commands/click#Arguments

<img width="1235" alt="Screen Shot 2021-03-31 at 10 47 30 AM" src="https://user-images.githubusercontent.com/1271364/113163823-85633180-920e-11eb-86b4-a134112bd49e.png">

**Comment by todd-m-kemp** [comment]
Hi @jennifer-shehane ! Thanks for taking a look.

I'm using `cypress-test-tiny` to reproduce and the only thing set in `cypress.json` is
```
{  "viewportHeight": 768,
  "viewportWidth": 1366
}
```
I've also been able to reproduce using 1920x1080.

Can you try either deleting the video in your account or adding a lot more? I don't recall how many videos my account had when I originally reported this but I just did some more testing and it seems to only happen when I have either no video content or a lot of video content. Here's a link to a video playlist showing the test fail in both these scenarios.

https://share.vidyard.com/watch/L3re4KvNpBuPMJhRBuAsuq

Let me know if there is anything else I can do to help with this.

**Comment by todd-m-kemp** [comment]
Hi @jennifer-shehane ! Have you had a chance to take another look at this issue? I am still able to reproduce the issue in version 7.0.0 this morning. My organization would love to be able to upgrade to version 7.0.0 in order to take advantage of the bug fixes and enhancements to `cy.intercept` but regrettably this issue prevents us from being able to do so.

**Comment by ulrik59** [comment]
Hello, I'm actually experiencing the same thing. Errors appeared at 6.7.1 version. See video:

- v6.7.0: https://www.loom.com/share/44c6dbcb113f4137a72d914b1ca7b147
- v6.7.1: https://www.loom.com/share/54b07c36ebcb4b20942f69886965d23c

I didn't have time to test v7.0.0+ yet.

**Comment by todd-m-kemp** [comment]
Thanks @ulrik59 for providing another example! I was having a hard time believing that I was the only one experiencing this problem. 😂 Hopefully Cypress can get it fixed as I would very much like to be able to upgrade beyond version 6.7.0.

**Comment by AdrianX19** [comment]
Unfortunately it does happen with v7.0.1 as well

**Comment by todd-m-kemp** [comment]
@jennifer-shehane suggested in an earlier comment that I look into the `scrollBehavior` option on `click` but initially I wasn't terribly interested since I didn't want to have to explicitly provide this option on all calls to `click` because this was a very broad problem causing failures in nearly every test of ours. However, yesterday I stumbled across the `scrollBehavior` option in `cypress.json` where you can set it globally. I did so, and this problem doesn't seem to occur in our test runs when using `center` as the `scrollBehavior` setting. Those of you experiencing this problem may also want to give that a try.

**Comment by jennifer-shehane** [comment]
@ulrik59 Yeah, your issue is definitely a bug. As evident in the video, Cypress scrolls back and forth forever until the click command times out. Can you provide a reproducible example?

@todd-m-kemp Yeah, this is what I would suggest. There are a few edge cases left where Cypress just doesn't guess the scrolling properly on click and I think setting your own preference is the best option.

**Comment by trev125** [comment]
@todd-m-kemp Thank you for adding that work around! That seems to have fixed things for me temporarily, and it was such a weird thing to see.

**Comment by farzadso** [comment]
@jennifer-shehane Even setting `scrollBehavior` doesn't work in my case. The only thing that works is `{ force: true }`.

Is there any other workaround for this?

**Comment by ChrisClsg** [comment]
Using `"scrollBehavior": "center"` fixes the problem for me as well.

**Comment by farzadso** [comment]
Just running the tests a few days after the issue arose (without any changes at all) fixed the problem. This is so weird.

**Comment by glomotion** [comment]
just chiming in here. I have noticed that after upgrading to cypress `9.x` - tests that have passed for thousands of test runs are now sometimes failing in CI due to the same error: 

```
Timed out retrying after 5050ms: cy.click() failed because this element:

<button data-testid="moo__clearButton" class="css-1ctlx4u-IconStyles-className-className-className-Icon-Button-ClearIcon">action_x</button>

is being covered by another element:

<button data-testid="moo__searchButton" class="css-xcpdcx-IconStyles-className-className-className-Icon-Button">action_...</button>

Fix this problem, or use {force: true} to disable error checking.
```

Incase someone from Cypress wants to take a look, you can see the test runs here: https://dashboard.cypress.io/projects/vyof1e/runs/416/test-results

**Comment by qWici** [comment]
Same bug. And I have no idea how it works. I have full-width & full-height modal wrapper with `position: fixed` with typical signin modal inside. I have a few simple test. They just input email & password and send form. If I change position prop of modal wrapper to absolute - tests passed. I tried to reproduce this bug on clean CRA app, but it worked fine.

**Comment by mgr32** [comment]
@jennifer-shehane Please see a repo with a minimal example demonstrating the issue: https://github.com/mgr32/cypress-issue-15618.
It breaks when style `:root { scroll-behavior: smooth }` (by default coming with Bootstrap) is present. 
Changing it to `:root {scroll-behavior: auto;}` or downgrading Cypress to 6.7.0 helps. 

As 6.7.1 is first version when it breaks, it's most probably caused by https://github.com/cypress-io/cypress/pull/15453.

**Comment by todd-m-kemp** [comment]
This problem is still very much reproducible. Some minor adjustments were needed to the code snippet (I've edited the original description accordingly) but I just reproduced the issue now in Cypress 12.8.1 in Chrome 113 on macOS 13.3.1.

**Comment by nagash77** [comment]
@thank you @todd-m-kemp for updating this example!  I will update the tags accordingly

**Comment by warrensplayer** [comment]
Possibly related: https://github.com/cypress-io/cypress/issues/20607
